libafl/src/stages/calibrate.rs

```diff
@@ -161,21 +196,38 @@ where
                 history_map.resize(map_len, O::Entry::default());
             }
 
-            for (first, (cur, history)) in
-                map_first.iter().zip(map.iter().zip(history_map.iter_mut()))
+            for (idx, (first, (cur, history))) in map_first
+                .iter()
+                .zip(map.iter().zip(history_map.iter_mut()))
+                .enumerate()
             {
                 if *first != *cur && *history != O::Entry::max_value() {
                     *history = O::Entry::max_value();
-                    unstable_entries += 1;
+                    unstable_entries.push(idx);
                 };
             }
 
             i += 1;
         }
 
         #[allow(clippy::cast_precision_loss)]
-        if unstable_entries != 0 {
-            *state.stability_mut() = Some((map_len - unstable_entries) as f32 / (map_len as f32));
+        if !unstable_entries.is_empty() {
+            // If we see new stable entries executing this new corpus entries, then merge with the existing one
+            if state.has_metadata::<UnstableEntriesMetadata>() {
+                let existing = state
+                    .metadata_mut()
+                    .get_mut::<UnstableEntriesMetadata>()
+                    .unwrap();
+                for item in unstable_entries {
+                    existing.unstable_entries.insert(item); // Insert newly found items
+                }
+                existing.map_len = map_len;
+            } else {
+                state.add_metadata::<UnstableEntriesMetadata>(UnstableEntriesMetadata::new(
+                    HashSet::from_iter(unstable_entries),
+                    map_len,
```

Comment:
is map_len what we want here?  IMO (unstable_edges / discovered_edges) would be better than (unstable_edges / all_edges).  As it is now undiscovered edges are assumed to be stable which can inflate the stability score, especially when the map is larger than the expected number of discovered edges.

Reply:
thanks for the suggestion
we already update the broker with (discovered_edges / all_edges), I feel it is fine to show stability as (unstable_edges / all_edges) right next to it. like this
```
[Stats       #7]  (GLOBAL) run time: 0h-0m-31s, clients: 9, corpus: 8401, objectives: 0, executions: 2505102, exec/sec: 81319
                  (CLIENT) corpus: 1089, objectives: 0, executions: 307364, exec/sec: 10062, edges: 2205/2219 (99%), stability: 434/2179 (19%)
[Stats       #8]  (GLOBAL) run time: 0h-0m-31s, clients: 9, corpus: 8401, objectives: 0, executions: 2505102, exec/sec: 81319
                  (CLIENT) corpus: 1098, objectives: 0, executions: 292780, exec/sec: 9618, edges: 2204/2219 (99%), stability: 417/2201 (18%)
```

Because these two numbers are calculated at different location, I have to either
1. In calibration stage, run through all the maps to count the non-0 element (but this will repeat the [same computation](https://github.com/AFLplusplus/LibAFL/blob/main/libafl/src/feedbacks/map.rs#L531) done in MaxMapfeedback::is_interesting, which is wastful of cpu poweres)
2. In is_interesting, update write the number of `discovered_edges` number into the metadata in state, but this function is a super hot path, personally I would not like to add additional overhead here..

Reply:
Yeah that makes sense.  It isn't really the job of the calibration stage to compute that and the overhead doesn't seem worth it. It would probably be best to do it in the monitor if it's done at all?  Because the monitor has all that data already.  but that's kinda special-case-y and might be best to just have people modify a monitor if they want it to display like that.  I do think that (unstable_edges / discovered_edges) is a more useful metric but that's admittedly rather subjective lol.

Reply:
With the dynamically adjustable map size, the difference is likely to be minimal anyway. But since it isn’t really a super precise metric anyway, and is only intended to give a rough idea, it feels like it’s probably fine as is. In any case the user can calculate the original style score with some simple maths from the output. It would be helpful to document what the metric is showing exactly and perhaps how it differs to AFL++ as I imagine many users will be transitioning from it to LibAFL and may expect it to be exactly the same, so may want to account for any differences to build confidence in making the switch.

Reply:
sure, I'll add docs tomorrow

Reply:
Awesome. Thanks.